ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartOptionsEditor.tsx

```diff
@@ -31,6 +46,22 @@ export function TimeSeriesChartOptionsEditor(props: TimeSeriesChartOptionsEditor
     );
   };
 
+  function updateLegendShow(show: boolean) {
+    onChange(
+      produce(value, (draft: TimeSeriesChartOptions) => {
+        draft.legend.show = show;
+      })
+    );
+  }
+
+  function updateLegendPosition(position: LegendPositionOptions['position']) {
```

Comment:
Tip: you might be able to avoid the cast below (i.e. `e.target.value as LegendPositionOptions['position']`) if you type this event handler using `SelectProps` and its generic argument. For example:

```tsx
const updateLegendPosition: SelectProps<LegendPosition>['onChange'] = (e) => {
  // Typescript will know e.target.value is of type LegendPosition or an empty string (which
  // can be ignored since you don't let people clear the value to empty string)
}
```

Reply:
hmm so I tried making [that change ](https://github.com/perses/perses/pull/660/commits/299ce400e573954d9dfb3ae0c5b484fa642206ce) and still getting `Type 'string' is not assignable to type '"bottom" | "right"'` error:
![image](https://user-images.githubusercontent.com/9369625/196758199-be898abc-9601-4cf7-a59f-3cbb8006a0ed.png)